mayastor-test/e2e/nightly/common/util.go

```diff
@@ -0,0 +1,361 @@
+			val := v.MapIndex(key)
+			switch sKey {
+			case "state":
+				msVol.state = val.Interface().(string)
+				break
+			case "reason":
+				msVol.reason = val.Interface().(string)
+				break
+			case "node":
+				msVol.node = val.Interface().(string)
+				break
+			}
+		}
+	}
+	return &msVol
+}
+
+// Check for a deleted Mayastor Volume,
+// the object does not exist if deleted
+func IsMSVDeleted(uuid string, dynamicClient * dynamic.Interface) bool {
+	msvGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorvolumes",
+	}
+
+	msv, err := (*dynamicClient).Resource(msvGVR).Namespace("mayastor").Get(context.TODO(), uuid, metav1.GetOptions{})
+
+	if err != nil {
+		// Unfortunately there is no associated error code so we resort to string comparison
+		if strings.HasPrefix(err.Error(), "mayastorvolumes.openebs.io") &&
+			strings.HasSuffix(err.Error(), " not found") {
+			return true
+		}
+	}
+
+	Expect(err).To(BeNil())
+	Expect(msv).ToNot(BeNil())
+	return false
+}
+
+// Check for a deleted Persistent Volume Claim,
+// either the object does not exist
+// or the status phase is invalid.
+func IsPVCDeleted(volName string, kubeInt * kubernetes.Interface) bool {
+	pvc, err := (*kubeInt).CoreV1().PersistentVolumeClaims("default").Get(context.TODO(), volName, metav1.GetOptions{})
+	if err != nil {
+		// Unfortunately there is no associated error code so we resort to string comparison
+		if strings.HasPrefix(err.Error(), "persistentvolumeclaims") &&
+			strings.HasSuffix(err.Error(), " not found") {
+			return true
+		}
+	}
+	// After the PVC has been deleted it may still accessible, but status phase will be invalid
+	Expect(err).To(BeNil())
+	Expect(pvc).ToNot(BeNil())
+	switch pvc.Status.Phase {
+	case
+		corev1.ClaimBound,
+		corev1.ClaimPending,
+		corev1.ClaimLost:
+		return false
+	default:
+		return true
+	}
+}
+
+// Check for a deleted Persistent Volume,
+// either the object does not exist
+// or the status phase is invalid.
+func IsPVDeleted(volName * string, kubeInt * kubernetes.Interface) bool {
+	vn := *volName
+	pv, err := (*kubeInt).CoreV1().PersistentVolumes().Get(context.TODO(), vn, metav1.GetOptions{})
+	if err != nil {
+		// Unfortunately there is no associated error code so we resort to string comparison
+		if strings.HasPrefix(err.Error(), "persistentvolumes") &&
+			strings.HasSuffix(err.Error(), " not found") {
+			return true
+		}
+	}
+	// After the PV has been deleted it may still accessible, but status phase will be invalid
+	Expect(err).To(BeNil())
+	Expect(pv).ToNot(BeNil())
+	switch pv.Status.Phase {
+	case
+		corev1.VolumeBound,
+		corev1.VolumeAvailable,
+		corev1.VolumeFailed,
+		corev1.VolumePending,
+		corev1.VolumeReleased:
+		return false
+	default:
+		return true
+	}
+}
+
+// Retrieve status phase of a Persistent Volume Claim
+func GetPvcClaimStatusPhase(volname string, kubeInt * kubernetes.Interface) (phase corev1.PersistentVolumeClaimPhase) {
```

Comment:
We could remove "Claim" from the name as we already use PVC.

Reply:
Fair enough